include/ddc/kernels/splines/spline_evaluator_2d_batched.hpp

```diff
@@ -0,0 +1,502 @@
+#include "spline_boundary_value.hpp"
+#include "spline_evaluator.hpp"
+#include "view.hpp"
+
+namespace ddc {
+
+template <
+        class ExecSpace,
+        class MemorySpace,
+        class BSplinesType1,
+        class BSplinesType2,
+        class interpolation_mesh_type1,
+        class interpolation_mesh_type2,
+        class... IDimX>
+class SplineEvaluator2DBatched
+{
+private:
+    // Tags to determine what to evaluate
+    struct eval_type
+    {
+    };
+
+    struct eval_deriv_type
+    {
+    };
+
+public:
+    using exec_space = ExecSpace;
+
+    using memory_space = MemorySpace;
+
+    using evaluator_type1
+            = ddc::SplineEvaluator<ExecSpace, MemorySpace, BSplinesType1, interpolation_mesh_type1>;
+    using evaluator_type2
+            = ddc::SplineEvaluator<ExecSpace, MemorySpace, BSplinesType2, interpolation_mesh_type2>;
+
+    using bsplines_type1 = typename evaluator_type1::bsplines_type;
+    using bsplines_type2 = typename evaluator_type2::bsplines_type;
+
+    using interpolation_domain_type1 = ddc::DiscreteDomain<interpolation_mesh_type1>;
+    using interpolation_domain_type2 = ddc::DiscreteDomain<interpolation_mesh_type2>;
+    using interpolation_domain_type
+            = ddc::DiscreteDomain<interpolation_mesh_type1, interpolation_mesh_type2>;
+
+    using vals_domain_type = ddc::DiscreteDomain<IDimX...>;
+
+    using bsplines_domain_type1 = ddc::DiscreteDomain<bsplines_type1>;
+    using bsplines_domain_type2 = ddc::DiscreteDomain<bsplines_type2>;
+    using bsplines_domain_type = ddc::DiscreteDomain<bsplines_type1, bsplines_type2>;
+
+    using batch_domain_type =
+            typename ddc::detail::convert_type_seq_to_discrete_domain<ddc::type_seq_remove_t<
+                    ddc::detail::TypeSeq<IDimX...>,
+                    ddc::detail::TypeSeq<interpolation_mesh_type1, interpolation_mesh_type2>>>;
+
+    template <typename Tag>
+    using spline_dim_type = std::conditional_t<
+            std::is_same_v<Tag, interpolation_mesh_type1>,
+            bsplines_type1,
+            std::conditional_t<std::is_same_v<Tag, interpolation_mesh_type2>, bsplines_type2, Tag>>;
+
+    using spline_domain_type =
+            typename ddc::detail::convert_type_seq_to_discrete_domain<ddc::type_seq_replace_t<
+                    ddc::detail::TypeSeq<IDimX...>,
+                    ddc::detail::TypeSeq<interpolation_mesh_type1, interpolation_mesh_type2>,
+                    ddc::detail::TypeSeq<bsplines_type1, bsplines_type2>>>;
+
+
+private:
+    spline_domain_type m_spline_domain;
+
+
+public:
+    explicit SplineEvaluator2DBatched(
+            spline_domain_type const& spline_domain,
+            SplineBoundaryValue<bsplines_type1> const&
+                    left1_bc, // Unused, to be restored in next MR
+            SplineBoundaryValue<bsplines_type1> const& right1_bc,
+            SplineBoundaryValue<bsplines_type2> const& left2_bc,
+            SplineBoundaryValue<bsplines_type2> const& right2_bc)
```

Comment:
```suggestion
            [[maybe_unused]] SplineBoundaryValue<bsplines_type1> const&
                    left1_bc, // Unused, to be restored in next MR
            [[maybe_unused]] SplineBoundaryValue<bsplines_type1> const& right1_bc,
            [[maybe_unused]] SplineBoundaryValue<bsplines_type2> const& left2_bc,
            [[maybe_unused]] SplineBoundaryValue<bsplines_type2> const& right2_bc)
```
to silence some warnings

Reply:
About assertions, I think those are all already present here:
https://github.com/CExA-project/ddc/blob/8f19b88f58522f9b75bcefb3435d147ebb48bdb1/include/ddc/kernels/splines/spline_builder_batched.hpp#L165C5-L166C94

Ofc it is less verbose than previous version because their are several instances od SplineBuilderBatched but I cannot see a situation where the a code would compile previously and would not with this new version.